bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/MenuItem.java

```diff
@@ -1121,27 +1124,47 @@ LRESULT wmCommandChild (long wParam, long lParam) {
 	return null;
 }
 
-LRESULT wmDrawChild (long wParam, long lParam) {
-	DRAWITEMSTRUCT struct = new DRAWITEMSTRUCT ();
-	OS.MoveMemory (struct, lParam, DRAWITEMSTRUCT.sizeof);
-	if (image != null) {
+LRESULT wmDrawChild(long wParam, long lParam) {
+	DRAWITEMSTRUCT struct = new DRAWITEMSTRUCT();
+	OS.MoveMemory(struct, lParam, DRAWITEMSTRUCT.sizeof);
+	if ((text != null || image != null)) {
 		GCData data = new GCData();
 		data.device = display;
 		GC gc = createNewGC(struct.hDC, data);
-		/*
-		* Bug in Windows.  When a bitmap is included in the
-		* menu bar, the HDC seems to already include the left
-		* coordinate.  The fix is to ignore this value when
-		* the item is in a menu bar.
-		*/
-		int x = (parent.style & SWT.BAR) != 0 ? MARGIN_WIDTH * 2 : struct.left;
-		Image image = getEnabled () ? this.image : new Image (display, this.image, SWT.IMAGE_DISABLE);
+
+		int x = (parent.style & SWT.BAR) == 0 ? MARGIN_WIDTH * 2 : struct.left;
 		int zoom = getZoom();
-		gc.drawImage (image, DPIUtil.scaleDown(x, zoom), DPIUtil.scaleDown(struct.top + MARGIN_HEIGHT, zoom));
-		if (this.image != image) image.dispose ();
-		gc.dispose ();
+		if (text != null) {
+			int flags = SWT.DRAW_MNEMONIC | SWT.DRAW_DELIMITER;
+			boolean isInactive = ((struct.itemState & OS.ODS_INACTIVE) != 0);
+			boolean isSelected = ((struct.itemState & OS.ODS_SELECTED) != 0);
+			int textMargin = 10; // left margin for x-coordinate spacing
```

Comment:
Please extract this into a static constant and give it a proper, self-explaining name that makes the comment obsolete (like `leftTextMargin`).